plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceData.java

```diff
@@ -0,0 +1,289 @@
+package com.cloud.hypervisor.xenserver.resource;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class CitrixResourceData {
```

Comment:
What is the need/justification for creating this data class? The data members were already part of appropriate XS resource classes. I don't see any benefit here from another level of indirection.